osc-server/src/main/java/org/osc/core/broker/service/persistence/SecurityGroupMemberEntityMgr.java

```diff
@@ -73,12 +86,22 @@ public static void fromEntity(SecurityGroupMember entity, SecurityGroupMemberIte
         Root<SecurityGroupMember> root = query.from(SecurityGroupMember.class);
 
         query = query.select(root).distinct(true)
-            .where(cb.equal(root.get("markedForDeletion"), false),
-                   cb.equal(root.get("securityGroup"), sg))
-            .orderBy(cb.asc(root.get("type")));
-
+                .where(cb.equal(root.get("markedForDeletion"), false),
+                       cb.equal(root.get("securityGroup"), sg))
+                .orderBy(cb.asc(root.get("type")));
 
         return em.createQuery(query).getResultList();
     }
 
+    private static void addVmPortInfo(SecurityGroupMemberItemDto dto, Set<VMPortDto> vmPortDtos, Set<VMPort> vmPorts) {
+        for (VMPort p : vmPorts) {
+            VMPortDto vmp = new VMPortDto();
```

Comment:
Create a constructor for vmportdto to take all the parameters instead of setting each of the properties